static/js/zamboni/files.js

```diff
@@ -1,41 +1,191 @@
 "use strict";
 
-if (typeof diff_match_patch !== 'undefined') {
-    diff_match_patch.prototype.diff_prettyHtml = function(diffs) {
-        /* An override of prettyHthml from diff_match_patch. This
-           one will not put any style attrs in the ins or del. */
-        var html = [];
-        for (var x = 0; x < diffs.length; x++) {
-            var op = diffs[x][0];    // Operation (insert, delete, equal)
-            var data = diffs[x][1];  // Text of change.
-            var lines = data.split('\n');
-            for (var t = 0; t < lines.length; t++) {
-                /* A diff gets an empty element on the end (the last \n).
-                   Unless the diff line in question does not have a new line on
-                   the end. We can't just set lines.length - 1, because this
-                   will just chop off lines. But if we don't trim these empty
-                   lines we'll end up with lines between each diff. */
-                if ((t + 1) == lines.length && lines[t] == '') {
-                    continue;
-                }
+$(function() {
+    // Compile our templates.
+    Highlighter.templates = _.object(_.map(
+        $('script[type="text/x-template"]'),
+        function(node) {
+            return [$.camelCase(node.id),
+                    _.template($(node).text())];
+        }));
+});
+
+var Highlighter = {
+    squash_space: function squash_space(str) {
+        // Squash non-significant whitespace in a string, so it can be
+        // ignored for the sake of diffs.
+        return str.replace(/[^\n]+/g, function (match) {
+            return match.trim()
+                .replace(/\s+/g, ' ')
+                .replace(/([^a-z0-9_$]) | (?![a-z0-9_$])/gi, "$1");
+        })
+    },
+
+    diff: function diff(left, right, brush) {
+        // Creates a unified diff of the arbitrary strings `left` and
+        // `right`, ignoring white-space changes within lines..
+
+        var differ = new diff_match_patch();
+
+        // Start by squashing the whitespace in both input strings,
+        // and converting them to arrays of characters.
+        var charred_inputs = differ.diff_linesToChars_(
+            this.squash_space(right), this.squash_space(left));
+
+        // Then have the diff library diff thos.
+        var diffs = differ.diff_main(charred_inputs[0],
+                                     charred_inputs[1], false);
+
+        // And convert them back to lines.
+        differ.diff_charsToLines_(diffs, charred_inputs[2]);
+
+        // If we have a brush, filter the left lines through the syntax
+        // highlighter. Otherwise, just escape them.
+        if (brush) {
+            var left_lines = this.highlight_lines(left, brush);
```

Comment:
Just a minor suggestion for better readability: I'd move the variable initialization to outside of the if statement:

```js
var foo;

if (bar) {
    foo = ...;
} else {
    foo = ...;
}
```

This makes the code more readable if the reader doesn't know or realize what hoisting is in JavaScript.